Save editor enhancements

Eblo opened the pull request:
- Add description tooltips to weekEventReg, eventInf, and owlActivation. Sourced from TCRF: https://tcrf.net/Proto:The_Legend_of_Zelda:_Majora's_Mask/Debug_Version/Event_Editor
  - weekEventReg checkboxes are color-coded based on the lifespan of the flag as indicated by that TCRF article (persistent, cycle, scene)
- Allow player to remove shield via editor, like sword
- Add ability to toggle Chateau magic


<img width="484" height="575" alt="image" src="https://github.com/user-attachments/assets/40c37c1c-a2c1-41fa-9440-e6c098584a41" />

<img width="186" height="90" alt="image" src="https://github.com/user-attachments/assets/e6f57073-f1b4-407e-9981-a01e97b46cd5" />

<img width="324" height="350" alt="image" src="https://github.com/user-attachments/assets/19ef386d-5ff5-4698-9abb-1fb50cfa064b" />


<!--- section:artifacts:start -->
### Build Artifacts
  - [2ship-linux.zip](https://nightly.link/HarbourMasters/2ship2harkinian/actions/artifacts/3890540936.zip)
  - [2ship-mac.zip](https://nightly.link/HarbourMasters/2ship2harkinian/actions/artifacts/3890543091.zip)
  - [2ship-windows.zip](https://nightly.link/HarbourMasters/2ship2harkinian/actions/artifacts/3890558950.zip)
<!--- section:artifacts:end -->

Eblo committed:
Save editor enhancements

Add tooltips to weekeventreg, owlActivation, and eventInf
Color code weekeventreg
Add toggle for Chateau Romani infinite magic
All shield to be removed

Eblo committed:
Disable clang format for flagTable

Eblo committed:
Preserve numerical tooltip

balloondude2 commented on mm/2s2h/DeveloperTools/SaveEditor.h:
```diff
@@ -1,6 +1,943 @@
+} FlagTable;
+
+// Reference https://tcrf.net/Proto:The_Legend_of_Zelda:_Majora's_Mask/Debug_Version/Event_Editor
+// The source was last referenced on 2025-08-05 and had a last updated value of 2024-06-29
```
This comments implies to me that this list should always come from tcrf. Is that what you intend, or do you see these potentially being updated as 2ship contributors document them?

Eblo replied:
I followed SoH's similar comment for its own flagtables:
https://github.com/HarbourMasters/Shipwright/blob/7b4df9bdb2ebde2bdd6581c10ffdc1aea4ee1573/soh/soh/Enhancements/debugger/debugSaveEditor.h#L28-L29

This isn't to mean that 2ship absolutely must source the TCRF article, but rather to document what source the flagtable used for its contents. In theory, that source could be updated in the future as unknown flags become understood and documented.

balloondude2 replied:
Great, just wanted to make sure.

balloondude2 commented on mm/2s2h/DeveloperTools/SaveEditor.cpp:
```diff
@@ -1822,25 +1827,26 @@ void DrawFlagsTab() {
             ImGui::EndGroup();
             break;
-        case 1: // weekEventReg
+        case WEEK_EVENT_REG:
```
With the persistent, cycle, scene/other coloring, I think we should add some kind of legend when the flag type is weekEventReg. 

Perhaps some dummy checkboxes at the top, with tooltips explaining the color?

Eblo replied:
A legend sounds like a good idea.

balloondude2 commented on mm/2s2h/DeveloperTools/SaveEditor.h:
```diff
@@ -1,6 +1,943 @@
+          { CYCLE_RESET, WEEKEVENTREG_91_80, "" },
+          { PERSISTENT, PACK_WEEKEVENTREG_FLAG(92, 0x01), "Started Gorman Bros. Race Once?" },
+          { CYCLE_RESET, PACK_WEEKEVENTREG_FLAG(92, 0x02), "Finished Gorman Bros. Race" },
+          { CYCLE_RESET, PACK_WEEKEVENTREG_FLAG(92, 0x04), "" },
```
92_01 is not persistent and these three flags are used together to form a 3-bit state to set/determine Gorman Bros Race states.

I would change these to SCENE_RESET and "Gorman Bros. Race State" or something.

balloondude2 commented on mm/2s2h/DeveloperTools/SaveEditor.h:
```diff
@@ -1,6 +1,943 @@
+          { CYCLE_RESET, WEEKEVENTREG_24_10, "Spoke to Goron Elder Twice" },
+          { CYCLE_RESET, WEEKEVENTREG_24_20, "" },
+          { CYCLE_RESET, WEEKEVENTREG_24_40, "Goron Elder Taught Lullaby Intro" },
+          { CYCLE_RESET, WEEKEVENTREG_24_80, "Spoke to Goron in Goron Shrine (Next to Baby, Right) Once" },
```
24_80 is PERSISTENT.

balloondude2 commented on mm/2s2h/DeveloperTools/SaveEditor.h:
```diff
@@ -1,6 +1,943 @@
+          { CYCLE_RESET, WEEKEVENTREG_RECEIVED_PENDANT_OF_MEMORIES, "Spoke to Kafei After Delivering Letter" },
+          { CYCLE_RESET, WEEKEVENTREG_DELIVERED_PENDANT_OF_MEMORIES, "Gave Pendant of Memories to Anju" },
+          { CYCLE_RESET, WEEKEVENTREG_51_02, "Letter to Kafei Related?" },
+          { SCENE_RESET, WEEKEVENTREG_51_04, "Hit Kafei's Bell" },
```
I'd call 51_04 SCENE_RESET, but thought it was interesting that this flag is set to persist.

Edit: Not asking for a change. Just pointing out something I thought was interesting

Eblo replied:
This flag gets cleared by the Kafei actor, making the persistent designation all the more perplexing.

balloondude2 reviewed (changes requested):
Left a couple comments after comparing the TCRF weekEventReg flag designation against `sPersistentCycleWeekEventRegs[]` from `z_sram_NES.c`.

Eblo committed:
Add color legend, update flagtable

Eblo commented:
Pushed an update addressing feedback. Here's the color legend for the weekeventreg flag table:

<img width="477" height="232" alt="image" src="https://github.com/user-attachments/assets/a6eb2b16-07ec-4722-9991-61fd37b221db" />

balloondude2 commented on mm/2s2h/BenGui/UIWidgets.cpp:
```diff
@@ -1147,7 +1186,8 @@ void DrawFlagArray8Mask(const std::string& name, uint8_t& flags, Colors color) {
         if (ImGui::IsItemHovered()) {
-            std::string label = fmt::format("0x{:02X} ({})", bitMask, flagIndex);
+            std::string label = WrappedText(flagEntry.description, 60).c_str();
+            label += fmt::format("{}0x{:02X} ({})", label.size() ? "\n" : "", flagIndex, flagIndex);
```
I'd like for the label to keep the bitMask, so we get something like `0x80 (7)` instead of `0x07 (7)`

```suggestion
            label += fmt::format("{}0x{:02X} ({})", label.size() ? "\n" : "", bitMask, flagIndex);
```

balloondude2 reviewed (changes requested):
These changes are looking really nice, just one last thing.

Eblo committed:
Retain bitmask

Co-authored-by: balloondude2 <55861555+balloondude2@users.noreply.github.com>
